test/gui/TestLogParser.py

```diff
@@ -32,29 +32,60 @@ def traverse_loop(data):
                                 if 'result' in error and error['result'] == 'PASS':
                                     continue
 
+                                # Again, we will have to loop through the 'tests' to get failing tests from Scenario Outlines.
+                                if "tests" in error and len(error['tests']) > 0:
+                                    for outlineStep in error['tests']:
+                                        if (
+                                            'result' in outlineStep
+                                            and outlineStep['result'] == 'ERROR'
+                                        ):
+                                            failing_test = {
+                                                "Feature File": str(
+                                                    feature_file['name']
+                                                ),
+                                                "Feature": str(feature['name']),
+                                                "Scenario": str(scenario['name']),
+                                                "Example": str(test_step['name']),
+                                                "Test Step": str(error['name']),
+                                                "Error Details": str(
+                                                    outlineStep['detail']
+                                                )
+                                                if ('detail' in outlineStep)
+                                                else "Error details not found",
+                                            }
+                                            failing_tests.append(failing_test)
+                                    continue
+
                                 # Append the information of failing tests into the list of failing tests
                                 # If the error detail is missing(occurs mainly in runtime error) then we display the entire error object.
-                                test = {
-                                    "Feature File": str(feature_file['name']),
-                                    "Feature": str(feature['name']),
-                                    "Scenario": str(scenario['name']),
-                                    "Test Step": str(test_step['name']),
-                                    "Error Details": str(error['detail'])
-                                    if ('detail' in error)
-                                    else "Error details not found",
-                                }
-
-                                failing_tests.append(test)
+                                if 'result' in error and error['result'] == 'ERROR':
```

Comment:
Is it always `ERROR`? Are you sure that there are no other value such as `FAIL` instead of `ERROR`?

Reply:
Yeah, It's always `ERROR` whenever the step fails.

Reply:
the comment says `If the error detail is missing(occurs mainly in runtime error) then we display the entire error object.` Where are you doing this here? And how does it look like (the final parsed log)? Can you comment the screenshot or something?

Reply:
I don't know exactly what that comment was for. Maybe it means to display `error` object rather than `Error details not found` whenever there is no `detail` in error.